src/api/qualicharge/migrations/versions/b69ed697b446_fix_admin_bound_types.py

```diff
@@ -0,0 +1,528 @@
+"""fix_admin_bound_types
+
+Revision ID: b69ed697b446
+Revises: 161e8a262e0a
+Create Date: 2024-10-08 16:04:02.172208
+
+"""
+
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+import geoalchemy2
+import sqlmodel
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision: str = "b69ed697b446"
+down_revision: Union[str, None] = "161e8a262e0a"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.alter_column(
+        "city",
+        "created_at",
+        existing_type=postgresql.TIMESTAMP(timezone=True),
+        nullable=False,
+    )
+    op.alter_column(
+        "city",
+        "updated_at",
+        existing_type=postgresql.TIMESTAMP(timezone=True),
+        nullable=False,
+    )
+    op.alter_column(
+        "city",
+        "id",
+        existing_type=sa.TEXT(),
+        type_=sa.Uuid(),
+        postgresql_using="id::uuid",
+        nullable=False,
+    )
+    op.alter_column(
+        "city",
+        "name",
+        existing_type=sa.TEXT(),
+        type_=sqlmodel.sql.sqltypes.AutoString(),
+        nullable=False,
+    )
+    op.alter_column(
+        "city",
+        "geometry",
+        existing_type=geoalchemy2.types.Geometry(
+            srid=4326,
+            from_text="ST_GeomFromEWKT",
+            name="geometry",
+            _spatial_index_reflected=True,
+        ),
+        nullable=False,
+    )
+    op.alter_column(
+        "city",
+        "population",
+        existing_type=sa.BIGINT(),
+        type_=sa.Integer(),
+        existing_nullable=True,
+    )
+    op.alter_column(
+        "city",
+        "area",
+        existing_type=sa.TEXT(),
+        type_=sa.Float(),
+        postgresql_using="area::float",
+        existing_nullable=True,
+    )
```

Comment:
I think we haven't a null area, perhaps we can change in `existing_nullable=False ` ?

Reply:
Unfortunately, we have few `NULL` entries at least for cities and departments:

```sql
SELECT
  city.name,
  city.area
FROM
  city
WHERE
  city.area IS NULL
```

| name                         | area |
| ---------------------------- | ---- |
| Marseille 1er Arrondissement |      |
| Marseille 2e Arrondissement  |      |
| Marseille 3e Arrondissement  |      |
| Marseille 4e Arrondissement  |      |
| Marseille 5e Arrondissement  |      |
| Marseille 6e Arrondissement  |      |
| Marseille 7e Arrondissement  |      |
| Marseille 8e Arrondissement  |      |
| Marseille 9e Arrondissement  |      |
| Marseille 10e Arrondissement |      |
| Marseille 11e Arrondissement |      |
| Marseille 12e Arrondissement |      |
| Marseille 13e Arrondissement |      |
| Marseille 14e Arrondissement |      |
| Marseille 15e Arrondissement |      |
| Marseille 16e Arrondissement |      |
| Lyon 1er Arrondissement      |      |
| Lyon 2e Arrondissement       |      |
| Lyon 3e Arrondissement       |      |
| Lyon 4e Arrondissement       |      |
| Lyon 5e Arrondissement       |      |
| Lyon 6e Arrondissement       |      |
| Lyon 7e Arrondissement       |      |
| Lyon 8e Arrondissement       |      |
| Lyon 9e Arrondissement       |      |
| Paris 1er Arrondissement     |      |
| Paris 2e Arrondissement      |      |
| Paris 3e Arrondissement      |      |
| Paris 4e Arrondissement      |      |
| Paris 5e Arrondissement      |      |
| Paris 6e Arrondissement      |      |
| Paris 7e Arrondissement      |      |
| Paris 8e Arrondissement      |      |
| Paris 9e Arrondissement      |      |
| Paris 10e Arrondissement     |      |
| Paris 11e Arrondissement     |      |
| Paris 12e Arrondissement     |      |
| Paris 13e Arrondissement     |      |
| Paris 14e Arrondissement     |      |
| Paris 15e Arrondissement     |      |
| Paris 16e Arrondissement     |      |
| Paris 17e Arrondissement     |      |
| Paris 18e Arrondissement     |      |
| Paris 19e Arrondissement     |      |
| Paris 20e Arrondissement     |      |
| Île de Clipperton            |      |

Reply:
J'aurai mieux fait de vérifier avant d'écrire ...